processing/src/main/java/org/apache/carbondata/processing/loading/DataLoadProcessBuilder.java

```diff
@@ -56,7 +56,7 @@ public AbstractDataLoadProcessorStep build(CarbonLoadModel loadModel, String[] s
       CarbonIterator[] inputIterators) throws Exception {
     CarbonDataLoadConfiguration configuration = createConfiguration(loadModel, storeLocation);
     SortScopeOptions.SortScope sortScope = CarbonDataProcessorUtil.getSortScope(configuration);
-    if (!configuration.isSortTable() || sortScope.equals(SortScopeOptions.SortScope.NO_SORT)) {
+    if (sortScope.equals(SortScopeOptions.SortScope.NO_SORT)) {
```

Comment:
I do not think this is the correct place to modify. You should correct the place where bad record logger is called